ui-core/src/main/java/com/adyen/checkout/components/ui/view/AdyenComponentView.kt

```diff
@@ -145,6 +146,25 @@ class AdyenComponentView @JvmOverloads constructor(
         if (pending) binding.progressBar.show() else binding.progressBar.hide()
     }
 
+    private fun Button.setText(
+        viewType: ComponentViewType,
+        componentParams: ComponentParams,
+        localizedContext: Context
+    ) {
+        if (viewType is AmountButtonComponentViewType && !componentParams.amount.isEmpty) {
```

Comment:
What happens if `viewType is AmountButtonComponentViewType` and  `amount` is empty?

Reply:
Since `AmountButtonComponentViewType` extends from `ButtonComponentViewType` it's gonna hold true for the `else if` case and then show the label coming from `viewType`